src/screens/claimables/ClaimingTransactionClaimable.tsx

```diff
@@ -0,0 +1,204 @@
+    const updatedTxPayload = { ...baseTxPayload, ...gasParams };
+
+    const gasLimit = await estimateGasWithPadding(updatedTxPayload, null, null, provider);
+
+    if (!gasLimit) {
+      updateTxFee(null, null);
+      logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to estimate gas limit'));
+      return;
+    }
+
+    if (needsL1SecurityFeeChains.includes(claimable.chainId)) {
+      const l1SecurityFee = await ethereumUtils.calculateL1FeeOptimism(
+        // @ts-expect-error - type mismatch, but this tx request structure is the same as in SendSheet.js
+        {
+          to: claimable.action.to,
+          from: accountAddress,
+          value: '0x0',
+          data: claimable.action.data,
+        },
+        provider
+      );
+
+      if (!l1SecurityFee) {
+        updateTxFee(null, null);
+        logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to calculate L1 security fee'));
+        return;
+      }
+
+      updateTxFee(gasLimit, null, l1SecurityFee);
+    } else {
+      updateTxFee(gasLimit, null);
+    }
+
+    setTxPayload({ ...updatedTxPayload, gasLimit });
+  }, [baseTxPayload, selectedGasFee, provider, claimable.chainId, claimable.action.to, claimable.action.data, updateTxFee, accountAddress]);
+
+  useEffect(() => {
+    if (baseTxPayload) {
+      estimateGas();
+    }
+  }, [baseTxPayload, estimateGas, selectedGasFee]);
+
+  const isTransactionReady = !!(isGasReady && isSufficientGas && isValidGas && txPayload);
+
+  const nativeCurrencyGasFeeDisplay = convertAmountToNativeDisplayWorklet(
+    selectedGasFee?.gasFee?.estimatedFee?.native?.value?.amount,
+    nativeCurrency,
+    true
+  );
+
+  const { mutate: claimClaimable } = useMutation({
+    mutationFn: async () => {
+      if (!txPayload) {
+        setClaimStatus('error');
+        logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to claim claimable due to missing tx payload'));
+        return;
+      }
+
+      const wallet = await loadWallet({
+        address: accountAddress,
+        showErrorIfNotLoaded: false,
+        provider,
+      });
+
+      if (!wallet) {
+        // Biometrics auth failure (retry possible)
+        setClaimStatus('error');
+        return;
+      }
+
+      const { errorMessage } = await walletExecuteRap(wallet, {
+        type: 'claimTransactionClaimableRap',
+        claimTransactionClaimableActionParameters: { claimTx: txPayload },
+      });
+
+      if (errorMessage) {
+        setClaimStatus('error');
+        logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to claim claimable due to rap error'), {
+          message: errorMessage,
+        });
+      } else {
+        setClaimStatus('success');
+        // Clear and refresh claimables data
+        queryClient.invalidateQueries(claimablesQueryKey({ address: accountAddress, currency: nativeCurrency }));
+        refetch();
+      }
```

Comment:
I don't think you need to invalidate and refetch. Invalidation should trigger a refetch.